analysis.ts

```diff
@@ -0,0 +1,147 @@
+// const { Analysis, Account, Utils } = require('@tago-io/sdk')
+// const axios = require('axios')
+// const { csv2jsonAsync } = require('json-2-csv')
+
+
+import { Account, Analysis, Utils } from "@tago-io/sdk";
+import { Data } from "@tago-io/sdk/out/common/common.types";
+import { TagoContext } from "@tago-io/sdk/out/modules/Analysis/analysis.types";
+import axios from "axios";
+import { csv2jsonAsync } from "json-2-csv";
+import { queue } from "async";
+
+// Fix the keys of the JSON object.
+function fixKeys(item: any) {
+  return Object.keys(item).reduce((final, key) => {
+    const new_key = key.replaceAll(' ', '_').toLowerCase()
+    final[new_key] = item[key]
+    return final
+  }, {} as any)
+}
+
+// Convert your CSV file to JSON.
+async function convertCSV(data_csv: string) {
+  const options = {
+    // use delimiter ; when exporting from excel
+    delimiter: {
+      field: ";"
+    },
+    trimFieldValues: true,
+    trimHeaderFields: true
+  }
+  const result = await csv2jsonAsync(data_csv, options)
+  return result.map((item) => fixKeys(item))
+}
+
+interface ICreateDevice {
+  account: Account
+  context: TagoContext
+  device: any
+  index: number
+  connector: string
+  network: string
+}
+
+// Create the device.
+async function createDevice({ account, context, device, index, connector, network }: ICreateDevice) {
+  if (!device.devicename) {
+    throw context.log(`Missing devicename, line ${index}`)
+  } else if (!device.devicetype) {
+    throw context.log(`Missing devicetype, line ${index}`)
+  } else if (device.type === "immutable" && !device.chunkperiod) {
+    throw context.log(`Missing chunkperiod, line ${index}`)
+  } else if (device.type === "immutable" && !device.chunkretention) {
+    throw context.log(`Missing chunkretention, line ${index}`)
+  }
+
+  await account.devices
+    .create({
+      name: device.devicename,
+      connector,
+      network,
+      type: device.devicetype,
+      chunk_period: device.chunkperiod,
+      chunk_retention: device.chunkretention,
+
+      // tags: [ { key: device.key, value: device.tag }],
+    })
+    .then(() => {
+      context.log(`Line ${index}: ${device.devicename} successfully created.`)
```

Comment:
change context.log to console.log

Also, is better to send a notification to the profile instead of just output to console.